### DISCUSSION TITLE
Is the JWT Claim Set in an Encrypted JWT from EncryptJWT class signed?

### DISCUSSION THREAD

**Discussion by langtutheky** [panva/jose#588] [Q&A] [answered]
Is an Encrypted JWT from EncryptJWT class contains a signed JWT claim? In other words, is the Encrypted JWT from EncryptJWT class behaves like a Nested JWT with a JWS as the plaintext of the JWE? This is because in the documentation of the jwtDecrypt function mentions the wordings `validate the JWT Claims Set` and `verify JWT`; which sounds like the JWT Claim Set is part of a JWS within the JWE (Encrypted JWT).

**Comment by panva** [comment, marked as answer]
There is no JWS inside the JWE when you use EncryptJWT.

> The EncryptJWT class is used to build and encrypt Compact JWE formatted JSON Web Tokens.

In other words, the JWT Claims Set ([intentionally worded like so](https://datatracker.ietf.org/doc/html/rfc7519#:~:text=and/or%20encrypted.-,JWT%20Claims%20Set,-A%20JSON%20object)) is the cleartext of the JWE. The result is a JSON Web Token (JWT) that is encoded in a **JWE**, not **JWS**.

**Reply by langtutheky** [reply]
Thank you, not sure if the documentations for jwtDecrypt function should not mention the term "verify" or "validate" if the purpose is to decrypt the cyphertext into plaintext and then parse the plaintext into JWT Claims Set. It would be more clear since "verify" and "validate" is commonly used in the context of signed JWT.

**Reply by panva** [reply]
> Verifies the JWT format (to be a JWE Compact format), decrypts the ciphertext, validates the JWT Claims Set.

I believe this is both clear and concise. I would be happy if you shared a complete suggestion.
